lib/connection.js

```diff
@@ -69,39 +72,55 @@ function Connection (options) {
 	}
 
 	util.extend(this.options, options);
-
-	this.configureAddress();
-
-	if (this.options.pfx || this.options.pfxData) {
+        this.configureAddress();
+        
+        if (this.options.pfx || this.options.pfxData) {
 		this.options.cert = options.cert;
 		this.options.key = options.key;
-	}
+        }
 
 	// Set cache length to 1 to ensure transmitted notifications can be sent.
 	this.options.cacheLength = Math.max(this.options.cacheLength, 1);
 	this.options.maxConnections = Math.max(this.options.maxConnections, 1);
-
+        this.options.minConnections = Math.max(this.options.minConnections, 1);
 	this.deferredConnection = null;
 	this.sockets = [];
 	this.notificationBuffer  = [];
-
 	this.socketId = 0;
-
 	this.failureCount = 0;
+        this.currentConnectionRoundRobin = 0;
 
 	// when true, we end all sockets after the pending notifications reach 0
 	this.shutdownPending = false;
 
-	// track when notifications are queued so transmitCompleted is only emitted one when
+	// track when notifications are queued so transmitCompleted is only emitted one when 
 	// notifications are transmitted rather than after socket timeouts
 	this.notificationsQueued = false;
 
 	this.terminated = false;
-
-	events.EventEmitter.call(this);
-}
+           
+        events.EventEmitter.call(this);
+        this.maintainMinConnection();
+}   
 
 sysu.inherits(Connection, events.EventEmitter);
+Connection.prototype.maintainMinConnection = function() {
+     var that = this;
```

Comment:
I don't approve of `var that = this` or similar. Please bind `createConnections` to `this`